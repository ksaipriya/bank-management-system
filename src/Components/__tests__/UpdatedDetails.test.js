import React from 'react';
import { render } from '@testing-library/react';
import Test from '../UpdatedDetails.js'
import Adapter from 'enzyme-adapter-react-16';
import {configure,ShallowWrapper,shallow} from 'enzyme';
import ShallowRenderer from 'react-test-renderer/shallow';

configure({ adapter: new Adapter() });
describe('Test case for testing updated details', () => {

    it('matches snapshot', () => {
        const shallowRenderer = new ShallowRenderer();
        shallowRenderer.render(<Test/>);
        const result = shallowRenderer.getRenderOutput();
        expect(result).toMatchSnapshot();
    });

    it('able to find an html element', () => {
        let shallowWrapper = new ShallowWrapper(<Test/>)
        expect(shallowWrapper.find('h4').html()).toContain('Update Details!');
    });

    it('proper rendering for name element', () => {
        const { getByText,getByLabelText } = render(<Test />)
        const name = getByText('Name');
        expect(name).toBeInTheDocument();
        const inputname = getByLabelText('Name');
        expect(inputname).toHaveAttribute('type', 'text')
    });

    it('proper rendering for password element', () => {
        const { getByText,getByLabelText } = render(<Test />)
        const password = getByText('Password')
        expect(password).toBeInTheDocument();
        const inputPassword = getByLabelText('Password');
        expect(inputPassword).toHaveAttribute('type', 'password')
    });

    it('proper rendering for email element', () => {
        const { getByText,getByLabelText } = render(<Test />)
        const name = getByText('Email');
        expect(name).toBeInTheDocument();
        const inputemail = getByLabelText('Email');
        expect(inputemail).toHaveAttribute('type', 'text')
    });

    it('proper rendering for Username element', () => {
        const { getByText,getByLabelText } = render(<Test />)
        const name = getByText('UserName');
        expect(name).toBeInTheDocument();
        const inputUname = getByLabelText('UserName');
        expect(inputUname).toHaveAttribute('type', 'text')
    });

    it('proper rendering for Guardian Type element', () => {
        const { getByText,getByLabelText } = render(<Test />)
        const name = getByText('Guardian Type');
        expect(name).toBeInTheDocument();
        const inputUname = getByLabelText('Guardian Type');
        expect(inputUname).toHaveAttribute('type', 'text')
    });

    it('proper rendering for Guardian Name element', () => {
        const { getByText,getByLabelText } = render(<Test />)
        const name = getByText('Guardian Name');
        expect(name).toBeInTheDocument();
        const inputUname = getByLabelText('Guardian Name');
        expect(inputUname).toHaveAttribute('type', 'text')
    });

    it('proper rendering for Address element', () => {
        const { getByText,getByLabelText } = render(<Test />)
        const name = getByText('Address');
        expect(name).toBeInTheDocument();
        const inputUname = getByLabelText('Address');
        expect(inputUname).toHaveAttribute('type', 'text')
    });

    it('proper rendering for Citizenship element', () => {
        const { getByText,getByLabelText } = render(<Test />)
        const name = getByText('Citizenship');
        expect(name).toBeInTheDocument();
        const inputUname = getByLabelText('Citizenship');
        expect(inputUname).toHaveAttribute('type', 'text')
    });

    it('proper rendering for Contact Number element', () => {
        const { getByText,getByLabelText } = render(<Test />)
        const name = getByText('Contact Number');
        expect(name).toBeInTheDocument();
        const inputUname = getByLabelText('Contact Number');
        expect(inputUname).toHaveAttribute('type', 'number')
    });

    it('proper rendering for Branch Name element', () => {
        const { getByText,getByLabelText } = render(<Test />)
        const name = getByText('Branch Name');
        expect(name).toBeInTheDocument();
        const inputUname = getByLabelText('Branch Name');
        expect(inputUname).toHaveAttribute('type', 'text')
    });

    it('proper rendering for Account Type element', () => {
        const { getByText } = render(<Test />)
        const name = getByText('Account Type');
        expect(name).toBeInTheDocument();
    });

    it('proper rendering for Deposit Amount element', () => {
        const { getByText,getByLabelText } = render(<Test />)
        const name = getByText('Deposit Amount');
        expect(name).toBeInTheDocument();
        const inputUname = getByLabelText('Deposit Amount');
        expect(inputUname).toHaveAttribute('type', 'number')
    });

    it('proper rendering for Gender element', () => {
        const { getByLabelText } = render(<Test />)
        const name = getByLabelText('Gender');
        expect(name).toBeInTheDocument();
    });

    it('proper rendering for Marital Status element', () => {
        const { getByLabelText } = render(<Test />)
        const name = getByLabelText('Marital Status');
        expect(name).toBeInTheDocument();
    });

    it('proper rendering for State element', () => {
        const { getByLabelText } = render(<Test />)
        const name = getByLabelText('State');
        expect(name).toBeInTheDocument();
    });

    it('proper rendering for Country element', () => {
        const { getByLabelText } = render(<Test />)
        const name = getByLabelText('Country');
        expect(name).toBeInTheDocument();
    });

    it('proper rendering for Identification Proof Type element', () => {
        const { getByText,getByLabelText } = render(<Test />)
        const name = getByText('Identification Proof Type');
        expect(name).toBeInTheDocument();
        const inputUname = getByLabelText('Identification Proof Type');
        expect(inputUname).toHaveAttribute('type', 'text')
    });

    it('proper rendering for Identification Document No. element', () => {
        const { getByText,getByLabelText } = render(<Test />)
        const name = getByText('Identification Document No.');
        expect(name).toBeInTheDocument();
        const inputUname = getByLabelText('Identification Document No.');
        expect(inputUname).toHaveAttribute('type', 'text')
    });

    it('proper rendering for Reference Account Holder Name element', () => {
        const { getByText,getByLabelText } = render(<Test />)
        const name = getByText('Reference Account Holder Name');
        expect(name).toBeInTheDocument();
        const inputUname = getByLabelText('Reference Account Holder Name');
        expect(inputUname).toHaveAttribute('type', 'text')
    });

    it('proper rendering for Reference Account Holder Acc Number element', () => {
        const { getByText,getByLabelText } = render(<Test />)
        const name = getByText('Reference Account Holder Acc Number');
        expect(name).toBeInTheDocument();
        const inputUname = getByLabelText('Reference Account Holder Acc Number');
        expect(inputUname).toHaveAttribute('type', 'text')
    });

    it('proper rendering for Reference Account Holder Address element', () => {
        const { getByText,getByLabelText } = render(<Test />)
        const name = getByText('Reference Account Holder Address');
        expect(name).toBeInTheDocument();
        const inputUname = getByLabelText('Reference Account Holder Address');
        expect(inputUname).toHaveAttribute('type', 'text')
    });

    it('Test Update Button', () => {
        const mockCallBack = jest.fn();
        const button = shallow(<button onClick={mockCallBack}>Update</button>);
        button.find('button').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });

    it('Test logout Button', () => {
        const mockCallBack = jest.fn();
        const button = shallow(<button onClick={mockCallBack}>Logout</button>);
        button.find('button').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });
})