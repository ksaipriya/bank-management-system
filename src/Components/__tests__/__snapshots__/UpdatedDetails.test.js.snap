// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for testing updated details matches snapshot 1`] = `
<Container
  fluid={false}
>
  <h4>
    Update Details!
  </h4>
  <Form
    className="login-form"
    inline={false}
  >
    <FormGroup>
      <FormLabel
        column={false}
        id="Name"
        srOnly={false}
      >
        Name
      </FormLabel>
      <br />
      <FormControl
        aria-labelledby="Name"
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="Name"
        type="text"
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        id="Email"
        srOnly={false}
      >
        Email
      </FormLabel>
      <br />
      <FormControl
        aria-labelledby="Email"
        id="email"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="text"
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        id="Password"
        srOnly={false}
      >
        Password
      </FormLabel>
      <br />
      <FormControl
        aria-labelledby="Password"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        id="Username"
        srOnly={false}
      >
        UserName
      </FormLabel>
      <br />
      <FormControl
        aria-labelledby="Username"
        id="username"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        type="text"
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        id="gtype"
        srOnly={false}
      >
        Guardian Type
      </FormLabel>
      <br />
      <FormControl
        aria-labelledby="gtype"
        id="guardiantype"
        name="guardianType"
        onChange={[Function]}
        placeholder="Guardian Type"
        type="text"
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        id="gname"
        srOnly={false}
      >
        Guardian Name
      </FormLabel>
      <br />
      <FormControl
        aria-labelledby="gname"
        id="guardianname"
        name="guardianName"
        onChange={[Function]}
        placeholder="Guardian Name"
        type="text"
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        id="address"
        srOnly={false}
      >
        Address
      </FormLabel>
      <br />
      <FormControl
        aria-labelledby="address"
        id="address"
        name="address"
        onChange={[Function]}
        placeholder="Address"
        type="text"
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        id="citizenship"
        srOnly={false}
      >
        Citizenship
      </FormLabel>
      <br />
      <FormControl
        aria-labelledby="citizenship"
        id="citizenship"
        name="citizen"
        onChange={[Function]}
        placeholder="Citizenship "
        type="text"
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        id="contact"
        srOnly={false}
      >
        Contact Number
      </FormLabel>
      <br />
      <FormControl
        aria-labelledby="contact"
        id="contact"
        name="contact"
        onChange={[Function]}
        placeholder="Contact Number"
        type="number"
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        id="bname"
        srOnly={false}
      >
        Branch Name
      </FormLabel>
      <br />
      <FormControl
        aria-labelledby="bname"
        id="branchname"
        name="branchName"
        onChange={[Function]}
        placeholder="Branch Name"
        type="text"
      />
    </FormGroup>
    <FormRow>
      <FormGroup
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
      >
        <FormLabel
          column={false}
          id="accounttype"
          srOnly={false}
        >
          Account Type
        </FormLabel>
        <br />
        <FormControl
          aria-labelledby="accounttype"
          as="select"
          onChange={[Function]}
        >
          <option
            value=""
          >
            AccountType
          </option>
          <option
            value="salary"
          >
            Salary
          </option>
          <option
            value="savings"
          >
            Savings
          </option>
        </FormControl>
      </FormGroup>
      <FormGroup
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
      >
        <FormLabel
          column={false}
          id="depositamount"
          srOnly={false}
        >
          Deposit Amount
        </FormLabel>
        <FormControl
          aria-labelledby="depositamount"
          id="deposit"
          name="depositAmt"
          onChange={[Function]}
          placeholder="Deposit Amount"
          readOnly={true}
          type="number"
        />
      </FormGroup>
    </FormRow>
    <FormRow>
      <FormGroup
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
      >
        <FormLabel
          column={false}
          id="gender"
          srOnly={false}
        >
          Gender
        </FormLabel>
        <FormControl
          aria-labelledby="gender"
          as="select"
          id="gender"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Select your Gender 
          </option>
          <option
            value="male"
          >
            Male
          </option>
          <option
            value="female"
          >
            Female
          </option>
        </FormControl>
      </FormGroup>
      <FormGroup
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
      >
        <FormLabel
          column={false}
          id="maritalStatus"
          srOnly={false}
        >
          Marital Status
        </FormLabel>
        <FormControl
          aria-labelledby="maritalStatus"
          as="select"
          id="maritalstatus"
          onChange={[Function]}
        >
          <option
            value=""
          >
            MaritalStatus
          </option>
          <option
            value="married"
          >
            Married
          </option>
          <option
            value="unmarried"
          >
            Unmarried
          </option>
        </FormControl>
      </FormGroup>
    </FormRow>
    <FormRow>
      <FormGroup
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
      >
        <FormLabel
          column={false}
          data-testid="Country"
          for="country"
          srOnly={false}
        >
          Country
        </FormLabel>
        <FormControl
          as="select"
          id="country"
          name="country"
          onChange={[Function]}
        >
          <option
            selected={true}
          >
             
          </option>
          <option>
            Afghanistan(AF)
          </option>
          <option>
            Aland Islands(AX)
          </option>
          <option>
            Albania(AL)
          </option>
          <option>
            Algeria(DZ)
          </option>
          <option>
            American Samoa(AS)
          </option>
          <option>
            Andorra(AD)
          </option>
          <option>
            Angola(AO)
          </option>
          <option>
            Anguilla(AI)
          </option>
          <option>
            Antarctica(AQ)
          </option>
          <option>
            Antigua And Barbuda(AG)
          </option>
          <option>
            Argentina(AR)
          </option>
          <option>
            Armenia(AM)
          </option>
          <option>
            Aruba(AW)
          </option>
          <option>
            Australia(AU)
          </option>
          <option>
            Austria(AT)
          </option>
          <option>
            Azerbaijan(AZ)
          </option>
          <option>
            Bahamas The(BS)
          </option>
          <option>
            Bahrain(BH)
          </option>
          <option>
            Bangladesh(BD)
          </option>
          <option>
            Barbados(BB)
          </option>
          <option>
            Belarus(BY)
          </option>
          <option>
            Belgium(BE)
          </option>
          <option>
            Belize(BZ)
          </option>
          <option>
            Benin(BJ)
          </option>
          <option>
            Bermuda(BM)
          </option>
          <option>
            Bhutan(BT)
          </option>
          <option>
            Bolivia(BO)
          </option>
          <option>
            Bonaire, Sint Eustatius and Saba(BQ)
          </option>
          <option>
            Bosnia and Herzegovina(BA)
          </option>
          <option>
            Botswana(BW)
          </option>
          <option>
            Bouvet Island(BV)
          </option>
          <option>
            Brazil(BR)
          </option>
          <option>
            British Indian Ocean Territory(IO)
          </option>
          <option>
            Brunei(BN)
          </option>
          <option>
            Bulgaria(BG)
          </option>
          <option>
            Burkina Faso(BF)
          </option>
          <option>
            Burundi(BI)
          </option>
          <option>
            Cambodia(KH)
          </option>
          <option>
            Cameroon(CM)
          </option>
          <option>
            Canada(CA)
          </option>
          <option>
            Cape Verde(CV)
          </option>
          <option>
            Cayman Islands(KY)
          </option>
          <option>
            Central African Republic(CF)
          </option>
          <option>
            Chad(TD)
          </option>
          <option>
            Chile(CL)
          </option>
          <option>
            China(CN)
          </option>
          <option>
            Christmas Island(CX)
          </option>
          <option>
            Cocos (Keeling) Islands(CC)
          </option>
          <option>
            Colombia(CO)
          </option>
          <option>
            Comoros(KM)
          </option>
          <option>
            Congo(CG)
          </option>
          <option>
            Cook Islands(CK)
          </option>
          <option>
            Costa Rica(CR)
          </option>
          <option>
            Cote D'Ivoire (Ivory Coast)(CI)
          </option>
          <option>
            Croatia(HR)
          </option>
          <option>
            Cuba(CU)
          </option>
          <option>
            Curaçao(CW)
          </option>
          <option>
            Cyprus(CY)
          </option>
          <option>
            Czech Republic(CZ)
          </option>
          <option>
            Democratic Republic of the Congo(CD)
          </option>
          <option>
            Denmark(DK)
          </option>
          <option>
            Djibouti(DJ)
          </option>
          <option>
            Dominica(DM)
          </option>
          <option>
            Dominican Republic(DO)
          </option>
          <option>
            East Timor(TL)
          </option>
          <option>
            Ecuador(EC)
          </option>
          <option>
            Egypt(EG)
          </option>
          <option>
            El Salvador(SV)
          </option>
          <option>
            Equatorial Guinea(GQ)
          </option>
          <option>
            Eritrea(ER)
          </option>
          <option>
            Estonia(EE)
          </option>
          <option>
            Ethiopia(ET)
          </option>
          <option>
            Falkland Islands(FK)
          </option>
          <option>
            Faroe Islands(FO)
          </option>
          <option>
            Fiji Islands(FJ)
          </option>
          <option>
            Finland(FI)
          </option>
          <option>
            France(FR)
          </option>
          <option>
            French Guiana(GF)
          </option>
          <option>
            French Polynesia(PF)
          </option>
          <option>
            French Southern Territories(TF)
          </option>
          <option>
            Gabon(GA)
          </option>
          <option>
            Gambia The(GM)
          </option>
          <option>
            Georgia(GE)
          </option>
          <option>
            Germany(DE)
          </option>
          <option>
            Ghana(GH)
          </option>
          <option>
            Gibraltar(GI)
          </option>
          <option>
            Greece(GR)
          </option>
          <option>
            Greenland(GL)
          </option>
          <option>
            Grenada(GD)
          </option>
          <option>
            Guadeloupe(GP)
          </option>
          <option>
            Guam(GU)
          </option>
          <option>
            Guatemala(GT)
          </option>
          <option>
            Guernsey and Alderney(GG)
          </option>
          <option>
            Guinea(GN)
          </option>
          <option>
            Guinea-Bissau(GW)
          </option>
          <option>
            Guyana(GY)
          </option>
          <option>
            Haiti(HT)
          </option>
          <option>
            Heard Island and McDonald Islands(HM)
          </option>
          <option>
            Honduras(HN)
          </option>
          <option>
            Hong Kong S.A.R.(HK)
          </option>
          <option>
            Hungary(HU)
          </option>
          <option>
            Iceland(IS)
          </option>
          <option>
            India(IN)
          </option>
          <option>
            Indonesia(ID)
          </option>
          <option>
            Iran(IR)
          </option>
          <option>
            Iraq(IQ)
          </option>
          <option>
            Ireland(IE)
          </option>
          <option>
            Israel(IL)
          </option>
          <option>
            Italy(IT)
          </option>
          <option>
            Jamaica(JM)
          </option>
          <option>
            Japan(JP)
          </option>
          <option>
            Jersey(JE)
          </option>
          <option>
            Jordan(JO)
          </option>
          <option>
            Kazakhstan(KZ)
          </option>
          <option>
            Kenya(KE)
          </option>
          <option>
            Kiribati(KI)
          </option>
          <option>
            Kosovo(XK)
          </option>
          <option>
            Kuwait(KW)
          </option>
          <option>
            Kyrgyzstan(KG)
          </option>
          <option>
            Laos(LA)
          </option>
          <option>
            Latvia(LV)
          </option>
          <option>
            Lebanon(LB)
          </option>
          <option>
            Lesotho(LS)
          </option>
          <option>
            Liberia(LR)
          </option>
          <option>
            Libya(LY)
          </option>
          <option>
            Liechtenstein(LI)
          </option>
          <option>
            Lithuania(LT)
          </option>
          <option>
            Luxembourg(LU)
          </option>
          <option>
            Macau S.A.R.(MO)
          </option>
          <option>
            Macedonia(MK)
          </option>
          <option>
            Madagascar(MG)
          </option>
          <option>
            Malawi(MW)
          </option>
          <option>
            Malaysia(MY)
          </option>
          <option>
            Maldives(MV)
          </option>
          <option>
            Mali(ML)
          </option>
          <option>
            Malta(MT)
          </option>
          <option>
            Man (Isle of)(IM)
          </option>
          <option>
            Marshall Islands(MH)
          </option>
          <option>
            Martinique(MQ)
          </option>
          <option>
            Mauritania(MR)
          </option>
          <option>
            Mauritius(MU)
          </option>
          <option>
            Mayotte(YT)
          </option>
          <option>
            Mexico(MX)
          </option>
          <option>
            Micronesia(FM)
          </option>
          <option>
            Moldova(MD)
          </option>
          <option>
            Monaco(MC)
          </option>
          <option>
            Mongolia(MN)
          </option>
          <option>
            Montenegro(ME)
          </option>
          <option>
            Montserrat(MS)
          </option>
          <option>
            Morocco(MA)
          </option>
          <option>
            Mozambique(MZ)
          </option>
          <option>
            Myanmar(MM)
          </option>
          <option>
            Namibia(NA)
          </option>
          <option>
            Nauru(NR)
          </option>
          <option>
            Nepal(NP)
          </option>
          <option>
            Netherlands(NL)
          </option>
          <option>
            New Caledonia(NC)
          </option>
          <option>
            New Zealand(NZ)
          </option>
          <option>
            Nicaragua(NI)
          </option>
          <option>
            Niger(NE)
          </option>
          <option>
            Nigeria(NG)
          </option>
          <option>
            Niue(NU)
          </option>
          <option>
            Norfolk Island(NF)
          </option>
          <option>
            North Korea(KP)
          </option>
          <option>
            Northern Mariana Islands(MP)
          </option>
          <option>
            Norway(NO)
          </option>
          <option>
            Oman(OM)
          </option>
          <option>
            Pakistan(PK)
          </option>
          <option>
            Palau(PW)
          </option>
          <option>
            Palestinian Territory Occupied(PS)
          </option>
          <option>
            Panama(PA)
          </option>
          <option>
            Papua new Guinea(PG)
          </option>
          <option>
            Paraguay(PY)
          </option>
          <option>
            Peru(PE)
          </option>
          <option>
            Philippines(PH)
          </option>
          <option>
            Pitcairn Island(PN)
          </option>
          <option>
            Poland(PL)
          </option>
          <option>
            Portugal(PT)
          </option>
          <option>
            Puerto Rico(PR)
          </option>
          <option>
            Qatar(QA)
          </option>
          <option>
            Reunion(RE)
          </option>
          <option>
            Romania(RO)
          </option>
          <option>
            Russia(RU)
          </option>
          <option>
            Rwanda(RW)
          </option>
          <option>
            Saint Helena(SH)
          </option>
          <option>
            Saint Kitts And Nevis(KN)
          </option>
          <option>
            Saint Lucia(LC)
          </option>
          <option>
            Saint Pierre and Miquelon(PM)
          </option>
          <option>
            Saint Vincent And The Grenadines(VC)
          </option>
          <option>
            Saint-Barthelemy(BL)
          </option>
          <option>
            Saint-Martin (French part)(MF)
          </option>
          <option>
            Samoa(WS)
          </option>
          <option>
            San Marino(SM)
          </option>
          <option>
            Sao Tome and Principe(ST)
          </option>
          <option>
            Saudi Arabia(SA)
          </option>
          <option>
            Senegal(SN)
          </option>
          <option>
            Serbia(RS)
          </option>
          <option>
            Seychelles(SC)
          </option>
          <option>
            Sierra Leone(SL)
          </option>
          <option>
            Singapore(SG)
          </option>
          <option>
            Sint Maarten (Dutch part)(SX)
          </option>
          <option>
            Slovakia(SK)
          </option>
          <option>
            Slovenia(SI)
          </option>
          <option>
            Solomon Islands(SB)
          </option>
          <option>
            Somalia(SO)
          </option>
          <option>
            South Africa(ZA)
          </option>
          <option>
            South Georgia(GS)
          </option>
          <option>
            South Korea(KR)
          </option>
          <option>
            South Sudan(SS)
          </option>
          <option>
            Spain(ES)
          </option>
          <option>
            Sri Lanka(LK)
          </option>
          <option>
            Sudan(SD)
          </option>
          <option>
            Suriname(SR)
          </option>
          <option>
            Svalbard And Jan Mayen Islands(SJ)
          </option>
          <option>
            Swaziland(SZ)
          </option>
          <option>
            Sweden(SE)
          </option>
          <option>
            Switzerland(CH)
          </option>
          <option>
            Syria(SY)
          </option>
          <option>
            Taiwan(TW)
          </option>
          <option>
            Tajikistan(TJ)
          </option>
          <option>
            Tanzania(TZ)
          </option>
          <option>
            Thailand(TH)
          </option>
          <option>
            Togo(TG)
          </option>
          <option>
            Tokelau(TK)
          </option>
          <option>
            Tonga(TO)
          </option>
          <option>
            Trinidad And Tobago(TT)
          </option>
          <option>
            Tunisia(TN)
          </option>
          <option>
            Turkey(TR)
          </option>
          <option>
            Turkmenistan(TM)
          </option>
          <option>
            Turks And Caicos Islands(TC)
          </option>
          <option>
            Tuvalu(TV)
          </option>
          <option>
            Uganda(UG)
          </option>
          <option>
            Ukraine(UA)
          </option>
          <option>
            United Arab Emirates(AE)
          </option>
          <option>
            United Kingdom(GB)
          </option>
          <option>
            United States(US)
          </option>
          <option>
            United States Minor Outlying Islands(UM)
          </option>
          <option>
            Uruguay(UY)
          </option>
          <option>
            Uzbekistan(UZ)
          </option>
          <option>
            Vanuatu(VU)
          </option>
          <option>
            Vatican City State (Holy See)(VA)
          </option>
          <option>
            Venezuela(VE)
          </option>
          <option>
            Vietnam(VN)
          </option>
          <option>
            Virgin Islands (British)(VG)
          </option>
          <option>
            Virgin Islands (US)(VI)
          </option>
          <option>
            Wallis And Futuna Islands(WF)
          </option>
          <option>
            Western Sahara(EH)
          </option>
          <option>
            Yemen(YE)
          </option>
          <option>
            Zambia(ZM)
          </option>
          <option>
            Zimbabwe(ZW)
          </option>
        </FormControl>
      </FormGroup>
      <FormGroup
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
      >
        <FormLabel
          column={false}
          data-testid="State"
          for="state"
          srOnly={false}
        >
          State
        </FormLabel>
        <FormControl
          as="select"
          id="state"
          name="state"
          onChange={[Function]}
        >
          <option
            selected={true}
          >
             
          </option>
        </FormControl>
      </FormGroup>
    </FormRow>
    <FormGroup>
      <FormLabel
        column={false}
        id="idtype"
        srOnly={false}
      >
        Identification Proof Type
      </FormLabel>
      <br />
      <FormControl
        aria-labelledby="idtype"
        id="idtype"
        name="idtype"
        onChange={[Function]}
        placeholder="Identification Proof Type"
        type="text"
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        id="iddoc"
        srOnly={false}
      >
        Identification Document No.
      </FormLabel>
      <FormControl
        aria-labelledby="iddoc"
        id="iddoc"
        name="iddoc"
        onChange={[Function]}
        placeholder="Identification Document No."
        type="text"
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        id="refname"
        srOnly={false}
      >
        Reference Account Holder Name
      </FormLabel>
      <FormControl
        aria-labelledby="refname"
        id="refname"
        name="refname"
        onChange={[Function]}
        placeholder="Reference Account Holder Name"
        type="text"
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        id="refnum"
        srOnly={false}
      >
        Reference Account Holder Acc Number
      </FormLabel>
      <FormControl
        aria-labelledby="refnum"
        id="refnum"
        name="refnum"
        onChange={[Function]}
        placeholder="Reference Account Holder Number"
        type="text"
      />
    </FormGroup>
    <FormGroup>
      <FormLabel
        column={false}
        id="refadd"
        srOnly={false}
      >
        Reference Account Holder Address
      </FormLabel>
      <FormControl
        aria-labelledby="refadd"
        id="refadd"
        name="refadd"
        onChange={[Function]}
        placeholder="Reference Account Holder Address"
        type="text"
      />
    </FormGroup>
    <FormRow>
      <FormGroup
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
      >
        <Button
          active={false}
          disabled={false}
          onClick={[Function]}
          variant="success"
        >
          Update
        </Button>
      </FormGroup>
      <FormGroup
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
      >
        <Button
          active={false}
          disabled={false}
          onClick={[Function]}
          variant="primary"
        >
          Logout
        </Button>
      </FormGroup>
    </FormRow>
  </Form>
</Container>
`;
